components/net/http_loader.rs

```diff
@@ -1637,40 +1588,108 @@ async fn http_network_or_cache_fetch(
         .await;
     }
 
-    // Step 11
+    // Step 15. If response’s status is 407, then:
     if let Some((StatusCode::PROXY_AUTHENTICATION_REQUIRED, _)) = response.status.as_ref() {
-        // Step 1
+        // Step 15.1 If request’s window is "no-window", then return a network error.
         if request_has_no_window {
             return Response::network_error(NetworkError::Internal(
                 "Can't find Window object".into(),
             ));
         }
 
-        // Step 2
-        // TODO: Spec says requires testing on Proxy-Authenticate headers
+        // (Step 15.2 does not exist, requires testing on Proxy-Authenticate headers)
 
-        // Step 3
-        // FIXME: Prompt the user for proxy authentication credentials
+        // FIXME: Step 15.3 If fetchParams is canceled, then return the appropriate network error for fetchParams.
```

Comment:
Again I think this could be done as mentioned above. TODO for another PR is good too.